### ISSUE TITLE
Unable to login into microsoft account

### ISSUE THREAD

**Issue by L0um15** [mindstorm38/portablemc#108]
**Describe the bug**
Title says it all, portablemc is unable to authenticate me whilst browser says it succeded.
I was already authenticated inside a browser since im using microsoft edge, i tried to remove permissions inside privacy settings but it didn't changed anything

`[Failed] auth.microsoft.incoherent_dat`

**To Reproduce**
PortableMC version: `3.1.1` `3.1.0`
Python version: `3.10.6`
Add-ons: `console`, `fabric`
Steps to reproduce the behavior: ...
```bash
portablemc start fabric:1.19.2 -m -l <email>
```

**Screenshots**

![1](https://user-images.githubusercontent.com/44427021/196993539-9857d191-f9b5-41b5-bd42-718593edaa5b.png)
![2](https://user-images.githubusercontent.com/44427021/196993592-d227e545-68a5-4fb4-ba8f-c33c208eaaf9.png)

**Comment by mindstorm38** [comment]
Two issues here:
- The typo, message should be `Incoherent authentication data, please retry.` not `auth.microsoft.incoherent_dat`
- The auth

**Comment by L0um15** [comment]
I retried 10 times already and it failed. Credentials are valid. I'll give it a try again when im back home

**Comment by mindstorm38** [comment]
Are you already logged on microsoft with this browser? And is this the same account as you're trying here? I suspect login.live.com to ignore the `login_hint` parameter (that preset the email you enter the CLI) when you already are logged on, this would not be that surprising... The only way of getting this error message is either to have the wrong nonce (if it exists, there is no reason for it to be wrong), or the wrong email, which could be the case if the authentication directly reply with another email.

To fix this, and I thought of this for a long time now, we could just type `pmc login -m` and we'll be able to enter any email.

**Comment by L0um15** [comment]
> Are you already logged on microsoft with this browser?

Yes i am, its microsoft edge and im using it for sync features.

I tried following commands:
```bash
portablemc login -m <email>
```
```
portablemc start fabric:1.19.2 -m -l <email>
```
yet still same thing happens.

One thing i haven't tried is to logout from browser and try to log in via portablemc like you suggested. I'll try it later.

**Comment by L0um15** [comment]
Unfortunately this didn't solved my problem

Here is what i tried so far:
1. I logged out from microsoft account in microsoft edge.
2. I removed my MS account from edge entirely
3. I removed app permissions from privacy settings inside ms account.
4. I tried to use another web browser (fresh instance of firefox)
5. I tried to use above commands.

Contents of the error: __incoherent_dat__, yet the browser says the opposite with "Logged in, you can close your tab".

I'am able to launch game in offline mode.

**Comment by mindstorm38** [comment]
Ok, thank you! Have you access to the source code, or can you modify it? If so, it would be interesting to check what contains the `id_token_payload` variable in `check_token_id(...)` method in the `MicrosoftAuthSession` class (`__init__.py`). 

```py
@classmethod
def check_token_id(cls, token_id: str, email: str, nonce: str) -> bool:
    id_token_payload = cls.decode_jwt_payload(token_id)
    print(f"{id_token_payload=}")
    return id_token_payload["nonce"] == nonce and id_token_payload["email"] == email
```

**Comment by mindstorm38** [comment]
Delayed to the next patch release until I get an answer, I can't reproduce this with my setup.

**Comment by L0um15** [comment]
Okay so i checked `check_token_id(...)` , 
```cs 
id_token_payload["nonce"] == nonce           // returns true
id_token_payload["email"] == email           // returns true (My email has uppercases) (Returns email with lowercases)
```
I tried to log in with only lowercases and it worked.

**Comment by mindstorm38** [comment]
Okay really interesting, I'll fix this

**Comment by JamiKettunen** [comment]
Ah I need to try this; my email also has uppercase letters and I've been unable to play on online mode servers so far and my username in-game has just been some "random" 8 character alpha-numeric string and not my username despite it showing up in `portablemc show auth` fine.

**Comment by JamiKettunen** [comment]
Ok update from my end: I just tried `portablemc start ... -m -l <email>` and that way I'm indeed logged in! Perhaps I've misunderstood the [session authentication](https://github.com/mindstorm38/portablemc#authentication-sessions) stuff altogether? I previously only did `portablemc login -m <email>`.

Actually related to the issue at hand, my host OS is (Void) Linux and the shell is `bash`, wonder if those could be something causing the difference in behavior.

**Comment by mindstorm38** [comment]
Should be fixed in next version, this version will be available in few days

**Comment by mindstorm38** [comment]
Re-open it if it doesn't fix your problems! 3.2.0 release in few minutes
